src/type_utils.jl

```diff
@@ -0,0 +1,9 @@
+"""
+    parameters(type)
+
+Extracts the type-parameters of the `type`.
+
+e.g. `parameters(Foo{A, B, C}) == [A, B, C]`
```

Comment:
Can't use a doctest here are i would need a struct to test it on.
COuld use a `Tuple` I guess, if one feels a doctest is important

Reply:
I'm fine with this as is